src/hooks/use-workspace-launcher.tsx

```diff
@@ -3,8 +3,16 @@ import { createUserAndWorkspaceAction, getWorkspaceUrlAction } from '@/server/ac
 import { notifications } from '@mantine/notifications'
 import { useState, useCallback } from 'react'
 
-const openWorkspaceInNewTab = (url: string) => {
-    window.open(url, 'child')
+interface OpenResult {
+    success: boolean
+    blocked: boolean
+    url: string
+}
+
+const openWorkspaceInNewTab = (url: string): OpenResult => {
+    const newWindow = window.open(url, 'child')
+    const blocked = !newWindow || newWindow.closed || typeof newWindow.closed === 'undefined'
```

Comment:
Nice popup detection! Didn't know that was possible

Reply:
I'm not 100% sure about the `closed`,  the MDN docs say that it might be the case that the window opens but is in a different security context.  I figured it'd be best to check it at least